aggregator/src/binaries/aggregator.rs

```diff
@@ -78,19 +81,41 @@ async fn run_aggregator(
         None,
     );
 
-    let garbage_collector_future = {
+    let garbage_collector_handle = {
         let datastore = Arc::clone(&datastore);
         let gc_config = config.garbage_collection.take();
         let meter = meter.clone();
         let stopper = stopper.clone();
-        async move {
+        spawn(async move {
             if let Some(gc_config) = gc_config {
+                info!("Running garbage collector");
                 run_garbage_collector(datastore, gc_config, meter, stopper).await;
             }
-        }
+        })
+    };
+
+    let key_rotator_handle = {
```

Comment:
While we've moved to a single-binary model to improve our build times, we are also moving away from any given process serving multiple roles (e.g. we don't want a single process that is both an aggregator & a garbage collector).

Is there a reason to run the key rotator as part of the aggregator, rather than making it a separate subcommand that we'd run as a separate deployment? I don't see any synchronization between the key rotator's work & the other pieces that might be running in-process; and IMO we don't want every aggregator to run key rotation (it's wasted effort, even in the likely case there are no concurrency bugs).

Reply:
Hmm. I guess it's because we think (small?) BYOH deployments might still want to run everything in a single process?

I guess this isn't the worst, but it will lead to pain when the BYOH deployment scales up & no longer wants to run GC/key rotation in the same process as the aggregators. For simplicity, I'd still somewhat prefer to factor things such that these are separate processes, but maybe that is controversial?

Reply:
> I don't see any synchronization between the key rotator's work & the other pieces that might be running in-process;

I'm going to implement synchronization in a future PR, i.e. the one that makes global HPKE keys required. The global HPKE keys cache will block until at least one global HPKE becomes available.

> Hmm. I guess it's because we think (small?) BYOH deployments might still want to run everything in a single process?

This is the case, yes. Similar in spirit to GC where our BYOH deployments run it in the same process. It's indeed wasteful for when many aggregator replicas are involved, but this I think this is a somewhat reasonable tradeoff in the interest of ease of deployment.

I think if we had a more prescriptive deployment solution for BYOH, i.e. a helm chart or `kustomize`-esque spec, I would be more comfortable omitting this functionality entirely, because we would be able to ship a kubernetes `CronJob` that invokes the key rotator.

n.b. the key rotator as implemented in the previous PR can (and should) be run in a separate process as a cronjob, for our non-BYOH deployments. Higher scale-out BYOH deployments can remove `key_rotator` from the `aggregator.yaml` config, and move it to a separate `key_rotator.yaml` config for running the key rotator separately.

Reply:
This sounds good; I guess it doesn't cost us much to _support_ running multiple logically-separate pieces in one process.